api/src/importers/era/base-record-utils.js

```diff
@@ -0,0 +1,157 @@
+'use strict';
+
+const mongoose = require('mongoose');
+const defaultLog = require('../../utils/logger')('alc-csv-base-record-utils');
+const RecordController = require('../../controllers/record-controller');
+
+/**
+ * ALC csv base record type handler that can be used directly, or extended if customizations are needed.
+ *
+ * @class BaseRecordUtils
+ */
+class BaseRecordUtils {
+  /**
+   * Creates an instance of BaseRecordUtils.
+   *
+   * @param {*} auth_payload user information for auditing
+   * @param {*} recordType an item from record-type-enum.js -> RECORD_TYPE
+   * @param {*} csvRow an array containing the values from a single csv row.
+   * @memberof BaseRecordUtils
+   */
+  constructor(auth_payload, recordType, csvRow) {
+    if (!recordType) {
+      throw Error('BaseRecordUtils - required recordType must be non-null.');
+    }
+
+    this.auth_payload = auth_payload;
+    this.recordType = recordType;
+    this.csvRow = csvRow;
+  }
+
+  /**
+   * Transform an alc-csv row into a NRPTI record.
+   *
+   * Note: Only transforms common fields found in ALL supported alc-csv types.
+   *       To include other values, extend this class and adjust the object returned by this function as needed.
+   *
+   * @param {object} csvRow alc-csv row (required)
+   * @returns {object} NRPTI record.
+   * @throws {Error} if record is not provided.
+   * @memberof BaseRecordUtils
+   */
+  transformRecord(csvRow) {
+    if (!csvRow) {
+      throw Error('transformRecord - required csvRow must be non-null.');
+    }
+
+    return {
+      _schemaName: this.recordType._schemaName,
+      recordType: this.recordType.displayName,
+
+      sourceSystemRef: 'era-csv'
```

Comment:
Not sure if it's a big deal, bu the ticket says source should be `nro-era-csv`

Reply:
was just following the pattern of the other importers, I can change it though

Reply:
It's probably okay to just leave it

Reply:
okok